src/test/regress/sql/remove_temp_schemas.sql

```diff
@@ -0,0 +1,49 @@
+CREATE OR REPLACE FUNCTION gp_execute_on_server(content int, query text)
+returns text language C as '$libdir/regress.so', 'gp_execute_on_server';
+
+-- First, handle master temp schemas
+DO $$
+DECLARE
+	nsp pg_namespace.nspname%type;
+BEGIN
+	FOR nsp IN
+		SELECT nspname
+		FROM (
+			SELECT nspname
+			FROM gp_dist_random('pg_namespace')
+			UNION
+			SELECT nspname
+			FROM pg_namespace
+		) n
+		LEFT JOIN pg_stat_activity ON
+			sess_id = regexp_replace(nspname, 'pg(_toast)?_temp_', '')::int
+		WHERE sess_id is null AND nspname ~ '^pg(_toast)?_temp_[1-9]{1}[0-9]*'
+	LOOP
+		EXECUTE format('DROP SCHEMA IF EXISTS %I CASCADE', nsp);
+	END LOOP;
+END $$;
+
+-- Then, handle segment temp schemas
```

Comment:
Is there an example where deleting from the coordinator is not enough?

Reply:
shemas may exists only at segments. and gpcheckat tests still checks the incinsistency

Reply:
> schemas may exists only at segments

drop such schema on coordinator does not drop it from segments?

Reply:
```
CREATE TEMP TABLE t1 (i int, j int);
SET debug_dtm_action_segment=2;
SET debug_dtm_action_target=protocol;
SET debug_dtm_action_protocol=commit_prepared;
SET debug_dtm_action=fail_begin_command;
DROP TABLE t1;

SELECT * FROM all_schemas WHERE nspname ~ '^pg(_toast)?_temp_[1-9]{1}[0-9]+';
     nspname      | gp_segment_id 
------------------+---------------
 pg_toast_temp_19 |             2
 pg_temp_19       |             2
(2 rows)

DROP SCHEMA pg_temp_19;
 DROP SCHEMA pg_temp_19;
ERROR:  schema "pg_temp_19" does not exist

```

quite obvious, that coordinator does not dispatch drop command if can't find the namespace in its catalog

Reply:
do we need such improvements on 6X?

Reply:
could you create new task for 6X?

Reply:
I mean, if temp schema is not dropped from corordinator in case of error (unlike current 7X behaviour), then original solution is enough to cover the issue on 6X.  common DROP SCHEMA IF EXISTS normally drops everithing

Reply:
Can temp schema be dropped on coordinator and not be dropped on segments in 6X?

Reply:
Yep, eventually it's possible. moreover, i've updated my gpdb build and the current case has the same behaviour as 7X case because of commit 66fbbe05bc591723c3911274ed8cc1c2f555340. I'll create the ticket

Reply:
Created the ticket for 6X